pyiron_base/job/generic.py

```diff
@@ -427,6 +436,12 @@ def refresh_job_status(self):
                 db=self.project.db,
                 job_id=self.job_id,
             )
+        elif state.database.database_is_disabled:
+            self._status = JobStatus(
+                initial_status=h5io.read_hdf5(
+                    self.project_hdf5.file_name, self.job_name + "/status"
+                )
+            )
 
```

Comment:
Same as my comment above, this could delegate to a `JobStatus.refresh` that then goes through this logic.